ndc-client/src/models.rs

```diff
@@ -454,20 +463,100 @@ pub enum BinaryComparisonOperator {
 }
 // ANCHOR_END: BinaryComparisonOperator
 
+// ANCHOR: ColumnSelector
+/// ColumnSelector consists of a column name (the first element) and possibly a 
+/// path to a field within a nested objects in that column.
+/// Multi-element ColumnSelectors are only valid for databases that support nested objects,
+/// e.g. MongoDB.
+#[derive(Clone, Debug, PartialEq)]
+pub struct ColumnSelector(pub Vec<String>);
```

Comment:
Have you considered representing this as a `(String, Vec<String>)` instead?

Reply:
Yeah, that might be a good idea.